core/services/workflows/v2/engine.go

```diff
@@ -153,6 +153,41 @@ func (e *Engine) start(ctx context.Context) error {
 	return nil
 }
 
+// nodeSyncLoop updates local node state each time a DON is received on the channel
+// as a DON update implies that the Cap Registry was synced and that the node
+// state should be refreshed.
+func (e *Engine) nodeSyncLoop(ctx context.Context, cleanup func(), ch <-chan capabilities.DON) {
+	defer cleanup()
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case _, open := <-ch:
+			if !open {
+				return
+			}
+
+			ctxwt, cancel := context.WithTimeout(ctx, time.Duration(e.cfg.LocalLimits.LocalNodeTimeoutMs)*time.Millisecond)
+			defer cancel()
+
+			localNode, err := e.cfg.CapRegistry.LocalNode(ctxwt)
+			if err != nil {
+				e.cfg.Lggr.Errorf("could not get local node state: %w", err)
+				e.cfg.Hooks.OnNodeSynced(localNode, err)
+				continue
+			}
+
+			e.cfg.Lggr.Debugw("Setting local node state",
+				"Workflow DON ID", localNode.WorkflowDON.ID,
+				"Workflow DON families", localNode.WorkflowDON.Families,
+				"Workflow DON Config Version", localNode.WorkflowDON.ConfigVersion,
+			)
+			e.cfg.Hooks.OnNodeSynced(localNode, nil)
+			e.localNode.Store(&localNode)
```

Comment:
These defers will accumulate with each tick until shutdown, but can just wrap it in a func to fix:
```suggestion
			func() {
				ctxwt, cancel := context.WithTimeout(ctx, time.Duration(e.cfg.LocalLimits.LocalNodeTimeoutMs)*time.Millisecond)
				defer cancel()
	
				localNode, err := e.cfg.CapRegistry.LocalNode(ctxwt)
				if err != nil {
					e.cfg.Lggr.Errorf("could not get local node state: %w", err)
					e.cfg.Hooks.OnNodeSynced(localNode, err)
					continue
				}
	
				e.cfg.Lggr.Debugw("Setting local node state",
					"Workflow DON ID", localNode.WorkflowDON.ID,
					"Workflow DON families", localNode.WorkflowDON.Families,
					"Workflow DON Config Version", localNode.WorkflowDON.ConfigVersion,
				)
				e.cfg.Hooks.OnNodeSynced(localNode, nil)
				e.localNode.Store(&localNode)
			} ()
```
(the linter normally catches these)

Reply:
Could avoid the unusual var names too then 🤷 
```go
func(ctx context.Context) {
	ctx, cancel := context.WithTimeout(ctx, time.Duration(e.cfg.LocalLimits.LocalNodeTimeoutMs)*time.Millisecond)
	defer cancel()

	localNode, err := e.cfg.CapRegistry.LocalNode(ctx)
	if err != nil {
		e.cfg.Lggr.Errorf("could not get local node state: %w", err)
		e.cfg.Hooks.OnNodeSynced(localNode, err)
		continue
	}

	e.cfg.Lggr.Debugw("Setting local node state",
		"Workflow DON ID", localNode.WorkflowDON.ID,
		"Workflow DON families", localNode.WorkflowDON.Families,
		"Workflow DON Config Version", localNode.WorkflowDON.ConfigVersion,
	)
	e.cfg.Hooks.OnNodeSynced(localNode, nil)
	e.localNode.Store(&localNode)
} (ctx)
```

Reply:
Ah sorry need to return a bool for continue too, but you get the idea